### ISSUE TITLE
Move `test_messages_documentation.py` to tests folder

### ISSUE THREAD

**Issue by jpy-git** [PyCQA/pylint#6170]
### Current problem

Currently the `test_messages_documentation.py` file is in the `doc` folder.

I usually like to run the full test suite via `pytest tests -n 8` before submitting a PR (because it's faster than waiting for CI).
However, this doesn't include `test_messages_documentation.py` since it's in a diff folder. Could we move it over to the tests folder for convenience?

### Desired solution

As above

**Comment by Pierre-Sassoulas** [comment]
I think it make sense to have all tests in tests/

**Comment by DanielNoord** [comment]
> Could we move it over to the tests folder for convenience?

My rational against this was that these "tests" are essentially testing the same thing as the other tests are doing as well. (Although the `check_messages` mess shows that this is not always the case).

I thought inflating local test time by running completely similar tests would be undesirable and considered this a pure CI test. Perhaps we could make an `addopt` like `--all-tests` or something to make it easier to run all of them? Running two separate tests that require a complete `Run` instance for every message we emit is quite time intensive.

**Comment by DudeNr33** [comment]
I am +1 for moving it to the ``tests``  directory. 
Another argument is that I cannot run the tests from the Test Explorer in VS code, which also means that I had to create a specific launch configuration for debugging ``test_messages_documentation``.

I also think that we should run them in CI as well. This way we can ensure that the examples are actually working as intended.

**Comment by DanielNoord** [comment]
> I also think that we should run them in CI as well. This way we can ensure that the examples are actually working as intended.

This is done with the `Checks / Documentation` job.

**Comment by Pierre-Sassoulas** [comment]
> This is done with the Checks / Documentation job.

Yeah, there's some additional work to consider when doing this issue, because having minimal check for documentation changes is really a game changer for documentation MR. Maybe a mark that we'd launch only inside the documentation CI job ?

**Comment by DudeNr33** [comment]
> > I also think that we should run them in CI as well. This way we can ensure that the examples are actually working as intended.
> 
> This is done with the `Checks / Documentation` job.

Sorry, misread your previous comment.
